cmd/agent/subcommands/run/command.go

```diff
@@ -369,6 +370,12 @@ func getSharedFxOption() fx.Option {
 		compressionimpl.Module(),
 		demultiplexerimpl.Module(),
 		dogstatsd.Bundle(),
+		fx.Provide(func(logsagent optional.Option[logsAgent.Component]) optional.Option[logsagentpipeline.Component] {
```

Comment:
Is this needed for the core agent? It looks like it's only used by the otel-agent.

Reply:
We're using logsagentpipeline now in the core agent collector that's being run for Agent OTLP ingest: https://github.com/DataDog/datadog-agent/pull/25032/files#diff-2ac4d5fcc57950600c6fecf01d2cd84a9a906d98c76089cbb78ab70a4d5549cdR50.

This way we can share the same logs pipeline across OTLP ingest, otel-agent, and the datadog exporter.

Reply:
Got it, thanks for the explanation!